components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/device/dao/DeviceFlowDAOImpl.java

```diff
@@ -196,6 +198,14 @@ public DeviceFlowDO getAuthenticationDetails(String deviceCode, String clientId)
                     authenticatedIDP = resultSet.getString(9);
                     scopes = getScopesForCodeId(resultSet.getString(10), connection);
 
+                    if (StringUtils.isBlank(subjectIdentifier)) {
+                        int idpId = UserSessionStore.getInstance().getIdPId(authenticatedIDP, tenantId);
+                        subjectIdentifier = UserSessionStore.getInstance().getUserId(userName, tenantId,
+                                    userDomain, idpId);
+                        log.info("Added userID as subject identifier as the subject identifier " +
+                                "column value is empty in the table");
```

Comment:
let's reword the log like "defaulting to unique user ID as the subject Identifier"

Reply:
Addressed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2129/commits/8f026912884901f0c6b00a64142a488140ec7c85